partiql-types/src/main/kotlin/org/partiql/value/PartiQLValueReader.kt

```diff
@@ -0,0 +1,10 @@
+package org.partiql.value
+
+import java.io.InputStream
+
+public interface PartiQLValueReader : AutoCloseable {
+
+    public fun readValue(input: InputStream): PartiQLValue
+
+    public fun readValues(input: InputStream): Iterator<PartiQLValue>
```

Comment:
We can move the input parameter out. 

Purposed API:
```
val reader = <builder>.standard().build(input)
val value = reader.readValue()
```

Reply:
Also, if we want to use the iterator style:
```
val reader = <builder>.standard().build(input)
reader.next()
val value = reader.readValue()
```

Reply:
This would not allow for reading multiple input streams with one reader. Classic example being initializing a reader with a dictionary do decompress various input streams.

Reply:
Most of the java reader I am familiar with does not take input stream as a parameter, which is why I purposed this change. Fundamentally, I want the reader API to align with Java's [reader](https://docs.oracle.com/javase/8/docs/api/java/io/Reader.html#read--).

If one truly desire to read multiple input stream with one reader, why not just pre-process the input stream?